src/TorchSharp/NN/Utils/PackedSequence.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        public static partial class nn
+        {
+            public static partial class utils
+            {
+                public static partial class rnn
+                {
+                    /// <summary>
+                    /// A packed batch of variable length sequences.
+                    /// </summary>
+                    public class PackedSequence : IDisposable
+                    {
+                        /// <summary>
+                        /// Class wrapping PyTorch's packedsequence object reference.
+                        /// </summary>
+                        internal sealed class HType : SafeHandle
+                        {
+                            public HType(IntPtr preexistingHandle, bool ownsHandle)
+                                : base(IntPtr.Zero, ownsHandle)
+                            {
+                                SetHandle(preexistingHandle);
+                            }
+
+                            public override bool IsInvalid => handle == IntPtr.Zero;
+
+                            // This is just for marshalling
+                            internal HType() : base(IntPtr.Zero, true)
+                            {
+                            }
+
+                            [DllImport("LibTorchSharp")]
+                            private static extern void THSNN_PackedSequence_dispose(HType handle);
+
+                            protected override bool ReleaseHandle()
+                            {
+                                THSNN_PackedSequence_dispose(this);
```

Comment:
>THSNN_PackedSequence_dispose [](http://example.com/codeflow?start=32&length=28)

should you assign the handle back to IntPtr.Zero? 

Also in the native side THSNN_PackedSequence_dispose would be nice to check for null/zero value before deleting it?

Reply:
There is a example how to use `SafeHandle` https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0 . And also it is aligned with `Optimizer` class. Let me double check how it behave.

Reply:
What I was trying to say is, in `internal sealed class HType : SafeHandle` you have been explicitly overriding `protected override void Dispose(bool disposing)`. This means when dispose is called the internal state of the safe handle will not be noticed as it is disposed. Look at the https://source.dot.net/#System.Private.CoreLib/SafeHandle.cs,98 to get the idea what the default implementation of the dispose does in the SafeHandle. You may not need to override `Dispose(bool disposing)` at all and let `SafeHandle` automaltically handle the state.  

Note, the sample you have pointed at https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0 is not overriding `Dispose(bool disposing)`.

Reply:
Thank you @tarekgh. Removed `Dispose(bool)` to let `SafeHandle` handle the state. Also `THSNN_PackedSequence_dispose()` receives `IntPtr` as `NativeMethods.CloseHandle` of https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0 does so.